staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1413,21 +1413,23 @@ func (e *Store) finalizeDelete(ctx context.Context, obj runtime.Object, runHooks
 // a matcher that matches by key. SelectionPredicate does this for you
 // automatically.
 func (e *Store) Watch(ctx context.Context, options *metainternalversion.ListOptions) (watch.Interface, error) {
+	if options == nil {
+		options = &metainternalversion.ListOptions{}
+	}
 	label := labels.Everything()
-	if options != nil && options.LabelSelector != nil {
+	if options.LabelSelector != nil {
 		label = options.LabelSelector
 	}
 	field := fields.Everything()
-	if options != nil && options.FieldSelector != nil {
+	if options.FieldSelector != nil {
 		field = options.FieldSelector
 	}
 	predicate := e.PredicateFunc(label, field)
 
 	resourceVersion := ""
-	if options != nil {
-		resourceVersion = options.ResourceVersion
-		predicate.AllowWatchBookmarks = options.AllowWatchBookmarks
-	}
+	resourceVersion = options.ResourceVersion
+	predicate.AllowWatchBookmarks = options.AllowWatchBookmarks
+
 	return e.WatchPredicate(ctx, predicate, resourceVersion, options.SendInitialEvents)
```

Comment:
```suggestion
	return e.WatchPredicate(ctx, predicate, options.ResourceVersion, options.SendInitialEvents)
```

Reply:
Thank you for the feedback! I've removed the redundant code and simplified it.